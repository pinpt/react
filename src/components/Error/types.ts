import { ReactElement, ReactNode } from 'react';
import { IActionLinkProps } from '../Internal/ActionLink';
import { ILogoProps } from '../Logo';

export interface IErrorProps {
	className?: string;
	logo?: ReactElement<ILogoProps | IActionLinkProps>;
	error?: string;
	title?: string;
	description?: string;
	link?: ReactNode;
}
