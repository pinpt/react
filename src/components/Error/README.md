### Displays an error message and code

[Full Docs](https://react.preview.pinpoint.com/?path=/docs/components-error) | [Github](https://github.com/pinpt/react/tree/master/src/components/Error)

#### Theming

The parent className for styling this component is `.Error`

<details>
	<summary>Base Theme Styles</summary>

```css
.Page.ErrorPage {
	@apply h-screen w-full flex flex-col justify-between items-stretch;
}

.errorWrapper {
	@apply flex-grow py-10 flex flex-col justify-center;
}

.Error {
	@apply flex flex-col justify-center items-center text-center;
}

.Error .logo {
	@apply mb-10 md:mb-14;
}

.Error .error {
	@apply uppercase font-medium;
	color: var(--page-error-color);
}

.Error .title {
	@apply my-4 text-4xl md:text-5xl font-semibold;
	color: var(--page-title-color);
}

.Error .description {
	@apply text-lg;
	color: var(--page-secondary-text-color);
}

.Error .link {
	@apply mt-10 flex text-xl items-center;
	color: var(--page-highlight-color);
}

.Error .link .icon {
	@apply ml-2;
}
```

</details>
