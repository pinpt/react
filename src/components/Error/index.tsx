import { ReactElement, ReactNode } from "react";
import { faArrowRight } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { ILogoProps } from "../Logo";
import { IActionLinkProps } from "../Internal/ActionLink";

export interface IErrorProps {
	className?: string;
	logo?: ReactElement<ILogoProps | IActionLinkProps>;
	error?: string;
	title?: string;
	description?: string;
	link?: ReactNode;
}

const Error = (props: IErrorProps) => {
	const {
		className = '',
		logo,
		error = 'Error',
		title = 'Something Went Wrong',
		description = 'Sorry, an unexpected error occurred.',
		link
	} = props;

	return (
		<div className={`Pinpoint Error ${className}`}>
			<div className="logo">{logo}</div>
			<div className="error">{error}</div>
			<h1 className="title">
				{title}
			</h1>
			<div className="description">{description}</div>
			<div className="link">
				{link ?? (
					<>
						<span className="text">Go to all posts</span>
						<FontAwesomeIcon icon={faArrowRight} className="icon" />
					</>
				)}
			</div>
		</div>
	)
}

export default Error;
