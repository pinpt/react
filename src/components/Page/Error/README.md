### A component to render a full error page.

[Full Docs](https://react.preview.pinpoint.com/?path=/docs/components-page-error) | [Github](https://github.com/pinpt/react/tree/master/src/components/Page/Error)

#### Theming

The parent className for styling this component is `.Page.Error`

<details>
	<summary>Base Theme Styles</summary>

```css
.Page.ErrorPage {
	@apply h-screen w-full flex flex-col justify-between items-stretch;
}

.errorWrapper {
	@apply flex-grow py-10 flex flex-col justify-center;
}
```

</details>
