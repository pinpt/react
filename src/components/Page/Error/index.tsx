import withWrapper from '../../Internal/withWrapper';

import { ReactElement } from 'react';
import { IErrorProps } from 'src/components/Error/types';

export interface IPageErrorProps {
	className?: string;
	error: ReactElement<IErrorProps>;
}

const Error = (props: IPageErrorProps) => {
	const { className = '', error } = props;

	return (
		<div className={`Pinpoint Page Error ${className}`}>
			{withWrapper(error, 'error')}
		</div>
	);
};

export default Error;
