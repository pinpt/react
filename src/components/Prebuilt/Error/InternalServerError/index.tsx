import Error from '../../../Error';
import Logo from '../../../Logo';
import PageError from '../../../Page/Error';
import Footer from '../../Footer';
import { IPrebuiltErrorProps } from '../types';

const InternalServerError = ({
	handleLinkClick,
	renderCopyright,
	renderFooter,
	renderLogo,
	renderSocial,
	renderSubscribe,
	site,
}: IPrebuiltErrorProps) => (
	<PageError
		error={
			<Error
				className="Prebuilt 500"
				logo={
					renderLogo?.(site) ?? (
						<Logo className="Prebuilt" src={site.logoUrl} href={site.theme?.logoLink ?? site.url} />
					)
				}
				error="500 Error"
				title="Internal Server Error"
				onClick={handleLinkClick}
			/>
		}
		footer={
			renderFooter?.(site) ?? (
				<Footer
					site={site}
					renderCopyright={renderCopyright}
					renderLogo={renderLogo}
					renderSocial={renderSocial}
					renderSubscribe={renderSubscribe}
				/>
			)
		}
	/>
);

export default InternalServerError;
