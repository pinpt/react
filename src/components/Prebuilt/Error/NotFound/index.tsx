import Error from '../../../Error';
import Footer from '../../Footer';
import Logo from '../../../Logo';
import PageError from '../../../Page/Error';

import { IPrebuiltErrorProps } from '../types';
import Pinpoint from '../../../Pinpoint';

const NotFound = ({
	handleLinkClick,
	renderCopyright,
	renderFooter,
	renderLogo,
	renderSocial,
	renderSubscribe,
	site,
}: IPrebuiltErrorProps) => (
	<Pinpoint siteId={site.id}>
		{() => {
			return (
				<PageError
					error={
						<Error
							className="Prebuilt 404"
							logo={
								renderLogo?.(site) ?? (
									<Logo className="Prebuilt" src={site.logoUrl} href={site.theme?.logoLink ?? site.url} />
								)
							}
							error="404 Error"
							title="Page Not Found"
							description="Sorry, we couldn’t find the page you’re looking for."
							onClick={handleLinkClick}
						/>
					}
					footer={
						renderFooter?.(site) ?? (
							<Footer
								site={site}
								renderCopyright={renderCopyright}
								renderLogo={renderLogo}
								renderSocial={renderSocial}
								renderSubscribe={renderSubscribe}
							/>
						)
					}
				/>
			);
		}}
	</Pinpoint>
);

export default NotFound;
