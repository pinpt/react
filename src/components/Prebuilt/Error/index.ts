import InternalServerError from './InternalServerError';
import NotFound from './NotFound';

export default {
	InternalServerError,
	NotFound,
};
