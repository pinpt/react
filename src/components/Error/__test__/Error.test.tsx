import renderer from 'react-test-renderer';
import Error from '..';

test('Test default', () => {
	const component = renderer.create(<Error />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom error', () => {
	const component = renderer.create(<Error error="Custom Error" />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom title', () => {
	const component = renderer.create(<Error title="Custom Title" />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom description', () => {
	const component = renderer.create(<Error description="Custom Description" />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom link', () => {
	const component = renderer.create(<Error link={<>Custom Link</>} />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
