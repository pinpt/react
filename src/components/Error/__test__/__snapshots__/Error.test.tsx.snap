// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test custom description 1`] = `
<div
  className="Pinpoint Error "
>
  <div
    className="logo"
  />
  <div
    className="error"
  >
    Error
  </div>
  <h1
    className="title"
  >
    Something Went Wrong
  </h1>
  <div
    className="description"
  >
    Custom Description
  </div>
  <div
    className="link"
  />
</div>
`;

exports[`Test custom error 1`] = `
<div
  className="Pinpoint Error "
>
  <div
    className="logo"
  />
  <div
    className="error"
  >
    Custom Error
  </div>
  <h1
    className="title"
  >
    Something Went Wrong
  </h1>
  <div
    className="description"
  >
    Sorry, an unexpected error occurred.
  </div>
  <div
    className="link"
  />
</div>
`;

exports[`Test custom link 1`] = `
<div
  className="Pinpoint Error "
>
  <div
    className="logo"
  />
  <div
    className="error"
  >
    Error
  </div>
  <h1
    className="title"
  >
    Something Went Wrong
  </h1>
  <div
    className="description"
  >
    Sorry, an unexpected error occurred.
  </div>
  <div
    className="link"
  >
    Custom Link
  </div>
</div>
`;

exports[`Test custom title 1`] = `
<div
  className="Pinpoint Error "
>
  <div
    className="logo"
  />
  <div
    className="error"
  >
    Error
  </div>
  <h1
    className="title"
  >
    Custom Title
  </h1>
  <div
    className="description"
  >
    Sorry, an unexpected error occurred.
  </div>
  <div
    className="link"
  />
</div>
`;

exports[`Test default 1`] = `
<div
  className="Pinpoint Error "
>
  <div
    className="logo"
  />
  <div
    className="error"
  >
    Error
  </div>
  <h1
    className="title"
  >
    Something Went Wrong
  </h1>
  <div
    className="description"
  >
    Sorry, an unexpected error occurred.
  </div>
  <div
    className="link"
  />
</div>
`;
